Makefile

```diff
@@ -222,6 +224,10 @@ ifeq ($(origin MONERO_ROOT), undefined)
   $(error You need to set an environment variable MONERO_ROOT \
     to your monero repository root)
 endif
+ifeq ($(origin MONERO_BUILD_ROOT), undefined)
+  $(error You need to set an environment variable MONERO_BUILD_ROOT \
+    to your monero build root)
+endif
```

Comment:
Logic is wrong. MONERO_BUILD_ROOT gets automatically set further up. Thus this cannot be reached.

Reply:
The only time it can ever run is if MONERO_ROOT was not set, and in this case, I wouldn't want a warning telling the user to set both MONERO_ROOT and MONERO_BUILD_ROOT, which would be inaccurate, as only MONERO_ROOT needs setting.

Reply:
With the changes in lines 39-46 the automatic discovery of MONERO_BUILD_ROOT is only used if it's not already set.

Reply:
Actually, you are right. The warning is redundant.
A better check would be for a compiled file expected to reside in MONERO_BUILD_ROOT.

Reply:
No. This is what this patch does for each case:

 1. Neither set. 2 errors messages.
 2. Just MONERO_ROOT set. MONERO_BUILD_ROOT  gets automatically set, no errors.
 3. Just MONERO_BUILD_ROOT. Error for MONERO_ROOT not being set.
 4. Both set. No error message.

You see yet how this section only occurs for bullet 1, when in fact it shouldn't prompt the user to set MONERO_BUILD_ROOT, as it is not a requirement (due to second bullet). 

From an ease of build perspective, just setting MONERO_ROOT is the _safe_ way. The Monero source will be in sync with what's been built. This is why I made this automatic in the first place.

Reply:
You want MONERO_BUILD_ROOT to reflect whats been built from MONERO_ROOT, which is a git repo. The automatic setting of MONERO_BUILD_ROOT is safest.

Reply:
Something along the lines of :
```
if (NOT EXISTS ${MONERO_BUILD_ROOT}/external/randomx/librandomx.a)
  $(error A required could not be found under MONERO_BUILD_ROOT)
endif
```

Reply:
No. It's not needed. The linker will complain of any missing libraries and the _existing_ automatic detection of the correct Monero build root is safest approach. The checked out Monero code and its built files.

Reply:
Still, the automatic detection only works when Monero was build in-source, and that's not always the case.

Reply:
Monero is not built "in-source" though. By default it is built to the `build` folder. What you're trying to say, I think, is Monero is not always built with the Monero Makefile, which always builds to a specific location (build/os/branch).

Reply:
Reference: https://github.com/monero-project/monero/pull/6036

Reply:
I don't see how that's related.

Reply:
In any case. Please squash commits.

Reply:
ty